### PR TITLE
Add Calibration logging for PEC

### LINKED ISSUE #1861: Logs for PEC Calibration
What happens to the logging in the case of PEC- what is the equivalent of this the following messages:
image

_Originally posted by @andreamari in https://github.com/unitaryfund/mitiq/pull/1845#pullrequestreview-1450484565_

### PR DESCRIPTION
**Pull request by Misty-W**
## Description

Add Calibration logging for PEC. 

Fixes #1861 

A mixture of ZNE and PEC strategies is also supported in a simple way, as the appropriate header is printed initially and when the current technique is different from the previous technique. 

---

### License

- [X] I license this contribution under the terms of the GNU GPL, version 3 and grant Unitary Fund the right to provide additional permissions as described in section 7 of the GNU GPL, version 3.

Before opening the PR, please ensure you have completed the following where appropriate.
- [x] I added unit tests for new code.
- [x] I used [type hints](https://www.python.org/dev/peps/pep-0484/) in function signatures.
- [X] I used [Google-style](https://google.github.io/styleguide/pyguide.html#383-functions-and-methods) docstrings for functions.
- [X] I [updated the documentation](../blob/master/docs/CONTRIBUTING_DOCS.md) where relevant.
- [X] Added myself / the copyright holder to the AUTHORS file

### CODE REVIEW

**Comment by Misty-W** [comment]
@natestemen, thought I'd clarify that this PR is ready for review now (after a few false starts, oops!)

File: mitiq/calibration/settings.py [hunk 1]
```diff
@@ -220,21 +221,33 @@ def to_dict(self) -> Dict[str, Any]:
...
+            row = (
+                performance,
+                circuit_type,
+                self.technique.name,
+                summary["representation_function"],
+                str(summary["operations"]),
+                str(summary["noise_level"]),
             )
-        )
+            print(row)
```
Comment (natestemen) [inline mitiq/calibration/settings.py:250]: When I run this with the `light_pec_settings` object defined in the tests I get
```
| performance | circuit | method | representation function | operations to mitigate | noise level |  qubit dependent |
| ----------- | ------- | ------ | ----------------------- | ---------------------- | ----------- | --------------- | 
('✅', 'mirror', 'PEC', <function represent_operation_with_local_depolarizing_noise at 0x2880f0940>, '[cirq.Circuit([\n    cirq.Moment(\n        cirq.CNOT(cirq.LineQubit(0), cirq.LineQubit(1)),\n    ),\n]), cirq.Circuit([\n    cirq.Moment(\n        cirq.CZ(cirq.LineQubit(0), cirq.LineQubit(1)),\n    ),\n])]', '0.001')
('❌', 'mirror', 'PEC', <function represent_operation_with_local_depolarizing_noise at 0x2880f0940>, '[cirq.Circuit([\n    cirq.Moment(\n        cirq.CNOT(cirq.LineQubit(0), cirq.LineQubit(1)),\n    ),\n]), cirq.Circuit([\n    cirq.Moment(\n        cirq.CZ(cirq.LineQubit(0), cirq.LineQubit(1)),\n    ),\n])]', '0.001')
```

I think we want to use the [`format`](https://docs.python.org/3/library/stdtypes.html#str.format) function like we do with ZNE above. We'll also want to make sure to use the function name so we don't print extraneous info like the memory address. Similarly, do we want to print the operations? Or is there a shorter way to represent them? I don't think putting them in a table like this is very readable.

File: mitiq/calibration/calibrator.py [hunk 2]
```diff
@@ -21,16 +21,25 @@
     ZNESettings,
     Strategy,
     BenchmarkProblem,
+    MitigationTechnique,
 )
 from mitiq.interface import convert_from_mitiq
 
-TABLE_HEADER_STR = (
+TABLE_HEADER_STR_ZNE = (
```
Comment (natestemen) [inline mitiq/calibration/calibrator.py:28]: I think the method should come first in the variable name now that there are two headers. So something like `ZNE_TABLE_HEADER`, WDYT?

File: mitiq/calibration/calibrator.py [hunk 3]
```diff
@@ -210,12 +222,22 @@ def run(self, log: bool = False) -> None:
...
                     warnings.simplefilter("ignore", UserWarning)
                     mitigated_value = strategy.mitigation_function(
                         circuit, expval_executor
                     )
+                if (
+                    log
+                    and count
+                    and strategy.technique
+                    is not self.strategies[count - 1].technique
+                ):
```
Comment (natestemen) [inline mitiq/calibration/calibrator.py:236]: I think for readability the last condition
```py
strategy.technique is not self.strategies[count - 1].techique
```
should be in parenthesis. Technically, I think it's not totally needed based on pythons [operator precedence](https://docs.python.org/3/reference/expressions.html#operator-precedence) (`is not` binds more tightly than `and` if I'm reading this correctly), but IMO it reads a little easier. Not totally necessary, but WDYT?

File: mitiq/calibration/calibrator.py [hunk 4]
```diff
@@ -210,12 +222,22 @@ def run(self, log: bool = False) -> None:
...
+                    if strategy.technique is MitigationTechnique.ZNE:
+                        print(TABLE_HEADER_STR_ZNE)
+                    elif strategy.technique is MitigationTechnique.PEC:
+                        print(TABLE_HEADER_STR_PEC)
```
Comment (natestemen) [inline mitiq/calibration/calibrator.py:240]: I think this is necessary for the moment, but I don't love how we now have to print table headers in two places (in the code). I feel like it's going to need to be refactored at some point to make all this work a little cleaner, but I'm not sure what that is now. Any thoughts?

**Review by natestemen — CHANGES_REQUESTED** [review]
Nice work, and good tests. Biggest request is getting the lines to print in a formatted way similar to ZNE, but most other stuff looks good.

File: mitiq/calibration/calibrator.py (hunk 3 above)
Comment (Misty-W) [reply mitiq/calibration/calibrator.py]: That makes sense, I've added the parentheses.

File: mitiq/calibration/calibrator.py (hunk 2 above)
Comment (Misty-W) [reply mitiq/calibration/calibrator.py]: Sounds good, I've updated the names accordingly.

File: mitiq/calibration/settings.py (hunk 1 above)
Comment (Misty-W) [reply mitiq/calibration/settings.py]: Good catch, thanks for pointing this out! 
I fixed the function name printing. 
As for the operations, I think users would want to track which operations are represented in the PEC calibration runs, even if that causes the table to be a bit bulky, but now the info is printed more compactly as operations (as intended).

File: mitiq/calibration/calibrator.py (hunk 4 above)
Comment (Misty-W) [reply mitiq/calibration/calibrator.py]: I tried removing the `if count` condition and the code for initial printing of the header, which I thought would be a more succinct equivalent, but then the first header wasn't printed so I had to revert back.

**Review by natestemen — CHANGES_REQUESTED** [review]
When running an example using the `light_pec_settings` from the tests it's printing out double headers.

Screenshot 2023-07-12 at 2 46 46 PM

The function name is also overflowing it's column. Do you think continuing to use a table here is the right choice if you want to print out the operations? I don't see a way that will fit in a text table.

**Comment by Misty-W** [comment]
> When running an example using the `light_pec_settings` from the tests it's printing out double headers.
> 
> Screenshot 2023-07-12 at 2 46 46 PM

Thanks for pointing this out! Not sure how I missed this behavior before, but it turned out that the count determining whether to print the header "initially" was resetting with every "problem" instance.

**Comment by Misty-W** [comment]
> The function name is also overflowing it's column. Do you think continuing to use a table here is the right choice if you want to print out the operations? I don't see a way that will fit in a text table.

I propose to keep the table and remove the operations, since all 2-qubit op's are represented by default now when running PEC calibration.

**Review by natestemen — APPROVED** [review]
Nice work Misty! Thanks for working to ensure everything fits into the table nicely.

**Pull request merged**